### DISCUSSION TITLE
How  to add a new gpu index, brute force like FLAT ,but save as int8 type for save gpu memory?  It's best to compute with gpu int8 . others is ok, like fp16,fp32.,because mainly to save gpu memory

### DISCUSSION THREAD

**Discussion by Hypoce** [milvus-io/milvus#28532] [Q&A and General discussion]
CPU: AMD64
GPU:A10

wanted feature: 
1. brute force like FLAT ,but save as int8 type for save gpu memory
2. compute with gpu int8. others is ok, like fp16,fp32. 
3. use multiple gpu with shard or replicate

we'v make:
develop at milvus@2.3.0  and make new index type and modify faiss sdk

**Comment by Hypoce** [comment]
additional information: mainly feature can run on multiple gpu bug have some bug
  1.entity num only 100w ,memory 10G
  2.each gpu's memory Uneven distribution
  3.peocess dump sometimes

**Comment by yhmo** [comment]
Currently, milvus only supports two kinds of GPU index: GPU_IVF_FLAT and GPU_IVF_PQ
If you want to add a new index type by your self, the major work is in the [knowhere](https://github.com/zilliztech/knowhere) project. The workload is pretty large, I don't think we have resources to support you at source code level. You may read the knowhere source code by yourself.

**Reply by Hypoce** [reply]
Maybe give some direction is very useful for me. now, we read milvus2.3.0, 2.3.2, 1.1.1 source code  ,2.3.0 not supports multiple gpu, 2.3.2 use nvidia raft lib, v1.1.1 is more complex.  which version is the small workload , including that three version but not limited to.
Mainly  wanted feature is miltiple gpu and int in gpu memory ,unnecessary compute with int8 ,can transform to fp16 or fp32 to compute.

**Reply by Hypoce** [reply]
There is a  wrong word in row 4， the correct statement is we wanted int8  in gpu memory

**Reply by yhmo** [reply]
It is a big feature, I have no idea. Our knowhere team can provide suggestions and guides, but we don't intend to involve them in this work.

**Reply by Hypoce** [reply]
Thanks anyway.
